source/common/network/address_impl.h

```diff
@@ -278,6 +288,42 @@ class EnvoyInternalInstance : public InstanceBase {
   EnvoyInternalAddressImpl internal_address_;
 };
 
+template <typename... Args>
+StatusOr<InstanceConstSharedPtr> createIpv4Instance(Args&&... args) {
```

Comment:
ah I didn't realize you just meant template methods. This is a great way to pass through the args to the ctor.

Let's take this one step further so you can reduce these 3 copies to one by templatizing the subtype (Ipv4Instance, Ipv6Instance, PipeInstance) and making the validators be static methods of the type. E.g.
```
template <class InstanceSubtype, typename... Args>
StatusOr<InstanceConstSharedPtr> createInstance(Args&&... args) {
  absl::Status status = InstanceSubtype::validate();
  if (!status.ok()) {
    return status;
  }
  std::shared_ptr<Ipv6Instance> instance =
      std::make_shared<Ipv6Instance>(status, std::forward<Args>(args)...);
  if (!status.ok()) {
    return status;
  }
  return instance;
}
```

Reply:
Thanks, will refactor.